src/components/table/ApprovalsTable.tsx

```diff
@@ -18,24 +20,49 @@ interface PaginationModel {
 
 const ApprovalsTable: React.FC<ApprovalsTableProps> = ({ view }) => {
   let columns: GridColDef[] = []
-  const { getPendingDocuments } = useDocumentApi()
-  // const [_open, setOpen] = useState(false)
+  const { getPendingDocuments, confirmDocument } = useDocumentApi()
+  const [open, setOpen] = useState(false)
   const [isLoading, setIsLoading] = useState(true)
   const [data, setData] = useState([])
+  const [documentId, setDocumentId] = useState('')
   const [rowCountState, setRowCountState] = useState<number>(0)
   const [paginationModel, setPaginationModel] = useState<PaginationModel>({
     page: 0,
     pageSize: 10
   })
 
-  const fetchData = async () => {
+  const handleClose = () => {
+    setOpen(false)
+  }
+
+  const handleScan = async (scanData: string | null) => {
+    setIsLoading(true)
+    if (scanData && scanData !== '') {
+      try {
+        const result = await confirmDocument({
+          id: documentId,
+          locationQRcode: scanData
+        })
+        console.log(result)
+        handleClose()
+        setIsLoading(false)
```

Comment:
Remove line 39, just need to `setIsLoading(true)`, fetchData auto `setIsLoading(false)` for you